types/denom_test.go

```diff
@@ -0,0 +1,77 @@
+package types
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	atom  = "atom"
+	uatom = "uatom"
+	matom = "matom"
+	katom = "katom"
+)
+
+func TestRegisterDenom(t *testing.T) {
+	unit := NewIntFromBigInt(new(big.Int).Exp(big.NewInt(10), big.NewInt(9), nil))
+	require.NoError(t, RegisterDenom("gwei", unit))
+	require.Error(t, RegisterDenom("gwei", unit))
+
+	res, ok := GetDenomUnit("gwei")
+	require.True(t, ok)
+	require.Equal(t, unit, res)
+
+	res, ok = GetDenomUnit("finney")
+	require.False(t, ok)
+	require.Equal(t, ZeroInt(), res)
+}
+
+func TestConvertCoins(t *testing.T) {
+	atomUnit := OneInt()
+	require.NoError(t, RegisterDenom(atom, atomUnit))
+
+	katomUnit := NewIntFromBigInt(new(big.Int).Exp(big.NewInt(10), big.NewInt(3), nil))
+	require.NoError(t, RegisterDenom(katom, katomUnit))
+
+	uatomUinit := NewIntFromBigInt(new(big.Int).Exp(big.NewInt(10), big.NewInt(6), nil))
+	require.NoError(t, RegisterDenom(uatom, uatomUinit))
+
+	matomUnit := NewIntFromBigInt(new(big.Int).Exp(big.NewInt(10), big.NewInt(5), nil))
```

Comment:
shouldn't matom be 10^-3? why 5 here? Also katom is -5

Reply:
These are just examples. I'll remove them to only show atom, uatom and matom

Reply:
Yes, this is what I pointed to in Slack some days ago. 
SI prefixes are so commonly used that anyone gets confused when it finds these odd exponents here. I would just clean this up and merge this PR.
It is just about cleaning up the test and good to go IMO.

Reply:
Something important is there is a test for prefixes such k.

1 katom = 1000 atom
1 atom = 0.001 katom

I would expect something like this to work in both directions...